modules/openid-federation-common/src/jvmMain/kotlin/com/sphereon/oid/fed/common/jwt/JoseJwt.jvm.kt

```diff
@@ -0,0 +1,54 @@
+package com.sphereon.oid.fed.common.jwt
+
+import com.nimbusds.jose.JWSAlgorithm
+import com.nimbusds.jose.JWSHeader
+import com.nimbusds.jose.JWSSigner
+import com.nimbusds.jose.JWSVerifier
+import com.nimbusds.jose.crypto.RSASSASigner
+import com.nimbusds.jose.crypto.RSASSAVerifier
+import com.nimbusds.jose.jwk.RSAKey
+import com.nimbusds.jose.jwk.gen.RSAKeyGenerator
+import com.nimbusds.jwt.JWTClaimsSet
+import com.nimbusds.jwt.SignedJWT
+import java.util.*
+
+actual fun sign(
+    payload: String,
+    opts: Map<String, Any>
+): String {
+    val rsaJWK = opts["key"] as RSAKey? ?: RSAKeyGenerator(2048)
+        .keyID(UUID.randomUUID().toString())
```

Comment:
@nklomp made a comment about not being possible to generate a random kid with uuid

Reply:
I've made header the second parameter of the sign function, the key should be passed in for both js and jvm now.